File: DIP/DIP_Lab_406/Lab/MatLab/Lab_02/streching.m
% Load the image
image_path = 'nature.jpeg';
image = imread(image_path);


normalized_image = im2double(image);


m = 0.5;
E = 10;


transformed_image = 1 ./ (1 + exp(-E * (normalized_image - m)));


output_image = uint8(transformed_image * 255);


figure;
subplot(1, 2, 1);
imshow(image);
title('Original Image');

subplot(1, 2, 2);
imshow(output_image);
title('Contrast Stretched Image');

File: DIP/DIP_Lab_406/Lab/Python/Lab_02/streching.py
import numpy as np
from PIL import Image
import matplotlib.pyplot as plt

def contrast_stretching(image, m, E):
    normalized_image = np.array(image) / 255.0
    transformed_image = 1 / (1 + np.exp(-E * (normalized_image - m)))
    output_image = (transformed_image * 255).astype(np.uint8)
    return output_image

image_path = 'Image/nature.jpeg'   
image = Image.open(image_path)   

m = 0.5   
E = 10   

output_image = contrast_stretching(image, m, E)

fig, axes = plt.subplots(1, 2, figsize=(10, 5))
axes[0].imshow(image)
axes[0].set_title('Original Image')
axes[0].axis('off')

axes[1].imshow(output_image)
axes[1].set_title('Contrast Stretched Image')
axes[1].axis('off')

plt.tight_layout()
plt.show()